crates/oci-distribution/src/client.rs

```diff
@@ -675,17 +709,27 @@ mod test {
     }
 
     #[tokio::test]
-    async fn test_pull_image() {
+    async fn test_pull() {
         for &image in TEST_IMAGES {
             let reference = Reference::try_from(image).expect("failed to parse reference");
 
             let image_data = Client::default()
-                .pull_image(&reference, &RegistryAuth::Anonymous)
+                .pull(
+                    &reference,
+                    &RegistryAuth::Anonymous,
+                    vec![manifest::WASM_LAYER_MEDIA_TYPE],
+                )
                 .await
                 .expect("failed to pull manifest");
 
             assert!(!image_data.layers.is_empty());
             assert!(image_data.digest.is_some());
+
+            // removing layer validation should fail
```

Comment:
This should be a separate test I think (and there should arguably be a test for if the client does accept certain media types but the image isn't any of those types).

Reply:
done